index.d.ts

```diff
@@ -0,0 +1,89 @@
+declare module "moleculer-apollo-server" {
+	import { ServiceSchema, Context } from "moleculer";
+	import { Config } from "apollo-server-core";
+	import { OptionsUrlencoded } from "body-parser";
+	import { SchemaDirectiveVisitor, IResolvers } from "graphql-tools";
+
+	export {
+		GraphQLUpload,
+		GraphQLExtension,
+		gql,
+		ApolloError,
+		toApolloError,
+		SyntaxError,
+		ValidationError,
+		AuthenticationError,
+		ForbiddenError,
+		UserInputError,
+		defaultPlaygroundOptions,
+	} from "apollo-server-core";
+
+	export * from "graphql-tools";
+
+	export interface ApolloServerOptions {
+		path: string;
+		disableHealthCheck: boolean;
+		onHealthCheck: () => {};
+	}
+
+	export class ApolloServer {
+		createGraphQLServerOptions(req: any, res: any): Promise<any>;
+		createHandler(options: ApolloServerOptions): void;
+		supportsUploads(): boolean;
+		supportsSubscriptions(): boolean;
+	}
+
+	export interface ActionResolverSchema {
+		action: string;
+		rootParams?: {
+			[key: string]: string;
+		};
+		dataLoader?: boolean;
+	}
+
+	export interface ServiceResolverSchema {
+		[key: string]: {
+			[key: string]: ActionResolverSchema;
+		};
+	}
+
+	export interface ServiceRouteOptions {
```

Comment:
A property that appears to be missing here is `mappingPolicy`.  I believe that should be defined as `mappingPolicy?: "all" | "restrict";`

Reply:
An additional missing property is `authentication?: boolean;`.

Reply:
You're right, I collected props from sample files and missed what you mentioned.